### PR TITLE
Python versions

### PR DESCRIPTION
**Pull request by david-i-berry**
Removal of end of life (or soon to be ) python versions from the tests.yml
Removal of numpy from requirements (installed by eccodes)

### CODE REVIEW

**Review by maaikelimper — APPROVED** [review]

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -7,11 +7,11 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```
Comment (tomkralidis) [inline .github/workflows/tests.yml:10]: Can we keep 3.8?

**Review by tomkralidis — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (RoryPTB) [reply .github/workflows/tests.yml:10]: Do we specifically need 3.8? It will be end of life later this year.

Comment (tomkralidis) [reply .github/workflows/tests.yml:10]: For anyone on focal, 3.8 is required.  We can remove a few months following 3.8 EOL.

**Comment by david-i-berry** [comment]
Changes now included in thread bug fix branch. Closing.

**Pull request closed without merging**